packages/data-stores/src/user/controls.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Internal dependencies
+ */
+import { wpcomRequest } from '../utils';
+import { CreateAccountAction } from './types';
+
+// @TODO: pass in locale, and remove magic strings for client_id and client_secret
+export default {
+	CREATE_ACCOUNT: async ( action: CreateAccountAction ) => {
+		try {
+			const request = {
+				path: '/users/new',
+				apiVersion: '1.1',
+				method: 'post',
+				body: {
+					email: action.params?.email,
+					is_passwordless: true,
+					signup_flow_name: 'gutenboarding',
+					validate: false,
+					client_id: '39911',
+					client_secret: 'cOaYKdrkgXz8xY7aysv4fU6wL6sK5J8a6ojReEIAPwggsznj4Cb6mW0nffTxtYT8',
```

Comment:
These usually come from `development.json` et al. Since Gutenberg is it's own app (I think @sirreal even wants to move it to `/app`) I guess it needs it's own app config? Doesn't seem like it's something it should share with Calypso.

Reply:
We don't even need anything as sophisticated as Calypso's config-per-environment. `client_id` and `client_secret` are the same in all environments, so just hard-coding them in a constants file should be fine.

To keep the config out of the package I guess they could just be included in `CreateAccountParams`? KISS?
It'd be cool if `User.register()` accepted parameters so the config could be passed in when initing the store.

Reply:
> Since Gutenberg is it's own app (I think @sirreal even wants to move it to /app) I guess it needs [its] own app config?

_Gutenboarding_ not _Gutenberg_, but you're quite right that I do want to move it to its own app to facilitate and clarify that it is isolated from the rest of Calypso.

We have made an exception for `config`, but this is at the border between Calypso and Gutenboarding and not what I'd consider to be "in" the Gutenboarding app:

https://github.com/Automattic/wp-calypso/blob/b2a01ea5deb23f03daad6042c3a6d0e2bac53a92/client/landing/gutenboarding/index.tsx#L7

<details>
<summary>Aside about separate app</summary

I think using the config makes perfect sense here. The Gutenboarding "app" would really be a React component ready to be rendered, and necessary config would be passed in from Calypso in `client/landing/gutenboarding`:

```tsx
import config from '../../config';
import { Gutenboard } from '../apps/gutenboard';

window.AppBoot = () => {
	if ( ! config.isEnabled( 'gutenboarding' ) ) {
		window.location.href = '/';
	} else {
		ReactDom.render( <Gutenboarding />, document.getElementById( 'wpcom' ) );
	}
};
```

</details>

In this particular case, it's important to keep in mind that we're not talking about a separate app, but the `@auotmattic/data-stores` package that is meant to be published and shared! It won't have its own configuration, but should receive configuration provided by consumers. We haven't had to deal with that yet 🤔 

I've just noticed your comment here about accepting parameters in this store's `register`: https://github.com/Automattic/wp-calypso/pull/38823/files#r367228911 and already reached the same conclusion I was about to suggest:

> It'd be cool if `User.register()` accepted parameters so the config could be passed in when initing the store.

I was going to suggest the same thing, it seems like the most sensible place. The drawback would be having to use configuration anywhere you want to use the store. The only way to get the store key is by invoking `register()` (or knowing it and using the string, but we'd like to discourage that 🙂). Additionally, it would be very concerning if different configurations were passed at different times, that could easily lead to surprising behavior 😱 

That could be mitigated in consuming code, but that would be up to consumers to get it right:

```ts
// user-store.ts
import values from 'config';
import { user } from '@automattic/data-stores';
export default user.register( values.foo );

// component.ts
import { useSelect } from '@wordpress/data';
import { USER_STORE } from './user-store';
function MyComponent() {
  const user = useSelect( select => select( USER_STORE ).getUser() );
  // …
}
```

Reply:
I've gone with passing in params to `User.register()` for this one.